WooCommerce/Classes/ViewRelated/Orders/Order Creation/OrderForm.swift

```diff
@@ -280,9 +280,9 @@ private struct ProductsSection: View {
                 }, content: {
                     ProductSelectorView(configuration: ProductSelectorView.Configuration.addProductToOrder,
                                     isPresented: $showAddProduct,
-                                    viewModel: viewModel.addProductViewModel)
+                                    viewModel: viewModel.handleProductsViewModel)
                         .onDisappear {
-                            viewModel.addProductViewModel.clearSearchAndFilters()
+                            viewModel.handleProductsViewModel.clearSearchAndFilters()
```

Comment:
As explained before, I am wondering if this has any effect now that we are creating a new view model any time we call the property.

Reply:
I removed the method call here: https://github.com/woocommerce/woocommerce-ios/pull/8943/commits/0be8ea5de00277c7a460532828643df66498dcb6 and added further tests.

Calling this method `.onDisappear` was necessary for clearing the search terms  when the `ProductSelectorViewModel` was lazily initialized, however now that we re-instantiate the object, the search terms are cleared by default. 

Within the same closure there's also an (unrelated to this feature) assignment `navigationButtonID = UUID()` to re-instantiate the button ID as needed, which we accomplish now automatically when the object is re-instantiated, so we shouldn't need this either, but I left it outside of the PR and will make a note to come back to it.